src/acp/policy/getAllowModes.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2021 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import type { ThingPersisted } from "../../interfaces";
+import type { AccessModes } from "../type/AccessModes";
+import { ACP } from "../constants";
+import { getModes } from "../internal/getModes";
+
+/**
+ * ```{note}
+ * The ACP specification is a draft. As such, this function is experimental and
+ * subject to change, even in a non-major release.
+ * See also: https://solid.github.io/authorization-panel/acp-specification/
+ * ```
+ *
+ * Policies allow or deny access modes over resources and their associated
+ * access control resource.
+ *
+ * @param policy The Policy Thing which allows retrieved access modes.
+ * @returns Policy URL array.
+ * @since 1.16.0
+ */
+export function getAllowModes<T extends ThingPersisted>(
```

Comment:
Is the generic necessary, since it isn't reflected in the return type ?

Reply:
We want to accept anything that happens to be a thing persisted, so for example if something had an extra property but happened to also be a ThingPersisted it wouldn't be accepted otherwise (or you'd have to cast, which is ugly and poluting the code).

<img width="484" alt="Screenshot 2021-11-19 at 15 32 12" src="https://user-images.githubusercontent.com/613619/142648736-8d2b39f1-a929-498a-aaad-8ff33d7e09f2.png">